rotkehlchen/chain/manager.py

```diff
@@ -984,6 +984,8 @@ def modify_blockchain_accounts(
         # we are adding/removing accounts, make sure query cache is flushed
         self.flush_cache('query_balances')
         self.flush_cache('query_balances', blockchain=blockchain)
+        self.flush_cache('query_balances', blockchain=None, ignore_cache=False, beaconchain_fetch_eth1=False)  # noqa: E501
+        self.flush_cache('query_balances', blockchain=blockchain, ignore_cache=False, beaconchain_fetch_eth1=False)  # noqa: E501
```

Comment:
Question: Why did you add this here(and every other location)?

This is me trying to understand why.

Reply:
This is due to the cache key is calculated from the function name + kwargs and these combinations of arguments might be passed from `rest.py` to the `query_balances` method, so we have to clear cache for these combinations

Reply:
Hmm interesting. Damn this is a really ugly thing we have here with these caches. Perhaps need to reconsider them.